src/aggregator/aggregator/elem_test.go

```diff
@@ -2390,6 +2390,190 @@ func TestGaugeFindOrCreateWithSourceSet(t *testing.T) {
 	require.Equal(t, 0, len(e.cachedSourceSets))
 }
 
+func TestExpireValues(t *testing.T) {
+	opts := newTestOptions()
+	resolutionDuration := opts.DefaultStoragePolicies()[0].Resolution().Window
+	resolution := xtime.UnixNano(resolutionDuration)
+	bufferPastDuration := opts.BufferForPastTimedMetricFn()(resolutionDuration)
+
+	valsNoGaps := []xtime.UnixNano{
+		resolution * 1,
+		resolution * 2,
+		resolution * 3,
+	}
+	valsGaps := []xtime.UnixNano{
+		resolution * 1,
+		resolution * 5,
+		resolution * 6,
+		resolution * 10,
+	}
+
+	for _, test := range []struct {
+		name             string
+		targetNanos      xtime.UnixNano
+		resendEnabled    bool
+		values           []xtime.UnixNano
+		expectedToExpire []xtime.UnixNano
+		expectedValues   []xtime.UnixNano
+	}{
+		// no gaps - resend disabled
+		{
+			"no gaps - resend disabled - zero target", resolution * 0, false, valsNoGaps, []xtime.UnixNano{}, valsNoGaps,
```

Comment:
seems like every name should be unique and more descriptive of what you are testing. it' s not really clear how test is different.

Reply:
Added more names. The main difference btwn the cases is the targetNanos we pass to expireVals